torch/csrc/jit/autodiff.cpp

```diff
@@ -500,6 +499,65 @@ static void liftConstants(Gradient& grad_desc, ReverseDetails& rev_info) {
   }
 }
 
+// we need to fold aten::_size_if_not_equal at the differentiation time
+// while we know the shapes of aten::_size_if_not_equal's arguments
+// Otherwise, they will become inputs to a reverse Graph, and we will
+// lose this information and we don't profile Scalars, or Lists yet.
+static void foldSizeIfNotEqual(Block *reverse_block) {
+
+  for (Node *top_node : reverse_block->nodes()) {
```

Comment:
This pass doesn't take into account the fact that the block may have nested blocks in it. I just went through and fixed liftConstants, which had the same limitation in https://github.com/pytorch/pytorch/pull/25361

Reply:
This comment does not appear to be addressed.